docs/_posts/2022-06-24-types-as-comments-proposal.md

```diff
@@ -0,0 +1,170 @@
+> 
+
+### Some examples
+
+The [proposal for types as comments][tc39-types-as-comments] will probably not confront anyone who has worked with either TypeScript or Flow with mind-bending concepts. Rather, the proposed syntax for types (and the associated features of the type system) very closely resembles that of TypeScript and Flow.
+
+A **type annotation** is written in the form `x: T` where `x` is a variable name and `T` is a type name. We may use type annotations in variable declarations or in function parameters.
+
+```typescript
+// a variable that stores a string
+let x: string; 
+// a function that takes two string parameters and returns a boolean value
+function equals(s: string, t: string): boolean { /* … */ }
+```
+
+Types may be **declared** using the `interface` keyword to introduce a new type or using the `type` keyword to declare aliases.
+
+```typescript
+// An object type with a properties coordinate and radius.
+interface Circle {
+  coordinate: Point;
+  radius: number
+}
+// A shorter alias for boolean
+type bool = boolean;
+```
+
+Classes also implicitly define types.
+
+```typescript
+// Implicitly declares a type Point
+class Point {
+  x: number;
+  y: number
+}
+```
+
+JavaScript's primitive data types have corresponding predefined types (such as `number` and `string`).
+
+Curiously, the proposal suggests ignoring anything in between the curly braces of both `interface` as well as `class` declarations. This does not make much sense to me at the moment, but will hopefully be cleared up in the upcoming drafts.
+
+Some more complicated type system features such as
+
+* * type unions (`type C = A | B`),
+* generics (`type A<T> = T[]`, `interface Box<T> { content: T; }`)
+* type and non-null assertions (`const point = JSON.parse("{\"x\": 0, \"y\": 0}") as Point` resp. `JSON.parse("{\"x\": 0, \"y\": 0}")!.x`)
+
+are also proposed to be supported in order to add some more expressiveness. 
+
+> :information_source: For the full list of features, take a look at [the current draft][tc39-types-as-comments].
+
+Lastly, an extension to the module system syntax is proposed so that types can be imported and exported as well using `import type` respectively `export type`.
+
+## Summary
+
+The types as comments proposal laid out by the TC39 technical committee for JavaScript suggests adding syntax for optional typing information to JavaScript. With these syntax additions, we may provide type information in JavaScript code to leverage static type checking just as in TypeScript or Flow in order to produce safer code. Having types as comments in JavaScript makes a lot of sense given the popularity of statically typed languages that compile to JavaScript such as TypeScript and Flow. In my opinion, including type information—without a default type checker—is a sensible and cautious approach and should definitely be considered.
```

Comment:
```suggestion
The types as comments proposal laid out by the TC39 technical committee for JavaScript suggests adding syntax for optional typing information to JavaScript. With these syntax additions, we may provide type information in JavaScript code to leverage static type checking just as in TypeScript or Flow in order to produce safer code. Having types as comments in JavaScript makes a lot of sense given the popularity of statically typed languages that compile to JavaScript such as TypeScript and Flow. In my opinion, including type information — without a default type checker — is a sensible and cautious approach and should definitely be considered.
```

Reply:
Spaces in between dashes is german grammer. In English there are none. See https://www.grammarly.com/blog/dash/.